gis/gdal/Portfile

```diff
@@ -292,69 +275,48 @@ pre-configure {
     }
 }
 
-post-configure {
-    foreach w ${worksrcpath_dirs} {
-        reinplace -E "s|(INST_PYMOD).*|\\1\t=\t${PYINST}|" \
-            ${w}/GDALmake.opt
-    }
-}
-
 # Set target to none
 build.target
 
+# Consider adding OpenCL variant with GDAL 3.7
+# see https://gdal.org/development/building_from_source.html#opencl
+# variant opencl description {Enable OpenCL} {
+#      configure.args-replace  -DGDAL_USE_OPENCL=OFF    -DGDAL_USE_OPENCL=ON
+# }
+
 variant lto description {Enable Link Time Optimization} {
-    configure.args-append   --enable-lto
+    configure.args-append   -DENABLE_IPO=ON
 }
 
-variant lzma \
-    description {Enable LZMA (7Z) compression support} {
+variant lzma description {Enable LZMA (7Z) compression support} {
     depends_lib-append      port:lzma
-    configure.args-delete   --without-liblzma
-    configure.args-append   --with-liblzma=yes
+    configure.args-replace  -DGDAL_USE_LIBLZMA=OFF    -DGDAL_USE_LIBLZMA=ON
 }
 
-variant libkml \
-    description {Enable libkml} {
-    depends_lib-append      port:libkml
-    configure.args-delete   --without-libkml
-    configure.args-append   --enable-driver-kmlsuperoverlay
-    configure.args-append   --with-libkml=${prefix}
-    configure.args-append   --with-libkml-inc=${prefix}/include
-    configure.args-append   --with-libkml-lib=${prefix}/lib
+variant cryptopp description {Enable Crypto++ support} {
+    depends_lib-append      port:libcryptopp
+    configure.args-replace  -DGDAL_USE_CRYPTOPP=OFF    -DGDAL_USE_CRYPTOPP=ON
 }
 
-variant grass description {Enable GRASS 7 GIS I/O} {
-    pre-configure {
-        if {[catch {lindex [registry_active grass7] 0}]} {
-            ui_error "The grass variant requires grass 7 to be installed."
-            ui_error "Since grass7 depends on gdal, please first install gdal WITHOUT the grass variant."
-            ui_error "Then install grass7, then reinstall gdal with the grass variant."
-            return -code error "Missing GRASS 7."
-        }
-    }
+variant libkml description {Enable libkml} {
+    PortGroup               boost 1.0
 
-    configure.args-delete   --without-grass
-    configure.args-append   --with-grass=${prefix}/share/grass78
+    depends_lib-append      port:libkml
+    boost.depends_type      build
```

Comment:
not sure exactly what is needed here, but the `boost` PG does set quite a few `cmake` related variables already (see [here](https://github.com/macports/macports-ports/blob/7d3cdee471ccc3514bea2eb1686c975249578a06/_resources/port1.0/group/boost-1.0.tcl#L147)). If the `gdal` port needs another variable name the PG probably will provide the necessary hints to do so.
Also, since you're setting the dependency type to `build` it would be good to check that nothing links to `boost` package (i.e., is required at runtime).

Reply:
It is not GDAL that uses boost, but libkml. The GDAL driver code includes libkml header files which in turn includes boost. GDAL expects the header files to be in standard `${prefix}/include` and has no way to know to look into eg.  `${prefix}/libexec/boost/1.76/include`.

Reply:
> Also, since you're setting the dependency type to `build` it would be good to check that nothing links to `boost` package (i.e., is required at runtime).


GDAL doesn't seem to link (as its not using boost directly):
```
otool -L /opt/local/lib/gdalplugins/ogr_LIBKML.dylib
/opt/local/lib/gdalplugins/ogr_LIBKML.dylib:
	@rpath/libgdal.32.dylib (compatibility version 32.0.0, current version 32.3.6)
	/opt/local/lib/libkmlbase.1.dylib (compatibility version 1.0.0, current version 1.3.0)
	/opt/local/lib/libkmldom.1.dylib (compatibility version 1.0.0, current version 1.3.0)
	/opt/local/lib/libkmlengine.1.dylib (compatibility version 1.0.0, current version 1.3.0)
	/opt/local/lib/libminizip.1.dylib (compatibility version 2.0.0, current version 2.0.0)
	/opt/local/lib/liburiparser.1.dylib (compatibility version 1.0.0, current version 1.0.30)
	/opt/local/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.13)
	/opt/local/lib/libexpat.1.dylib (compatibility version 10.0.0, current version 10.10.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
	/opt/local/lib/openmpi-mp/libmpi_cxx.40.dylib (compatibility version 71.0.0, current version 71.1.0)
	/opt/local/lib/openmpi-mp/libmpi.40.dylib (compatibility version 71.0.0, current version 71.4.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1300.23.0)
```

Reply:
> It is not GDAL that uses boost, but libkml. The GDAL driver code includes libkml header files which in turn includes boost. GDAL expects the header files to be in standard `${prefix}/include` and has no way to know to look into eg. `${prefix}/libexec/boost/1.76/include`.

that sounds strange to me... if `gdal` doesn't use `boost` then it shouldn't need to have it as a built dependency. If the `libkml` port uses `boost` then it should probably specify in that port where the boost headers are and `gdal` should be able to use those - or am I missing something here?

Reply:
> that sounds strange to me... if `gdal` doesn't use `boost` then it shouldn't need to have it as a built dependency. 

There's no need for `GDAL` to have `boost` as dependency. With the GDAL variant (and port) `libkml`, which *does* have `boost` as dependency, however, GDAL's libkml-driver hooks to the libkml API (which in turn does, correctly, include boost API). libkml know where to look for boost header files, but GDAL doesn't. This would be all good and fine if they were located in "standard" location for include files (i.e. `${prefix}/include`) .

> If the `libkml` port uses `boost` then it should probably specify in that port where the boost headers are and `gdal` should be able to use those - or am I missing something here?

The boost PG defines the version used and ultimately the location of the include files. This is the reason, the only solution I could come up with, was the present by adding the boost PG to GDAL variant libkml. Note this only affects GDAL with the libkml variant. If there is a way to get the include path to boost, from boost PG, without adding the PG to GDAL, that would be an alternative and perhaps better solution.

Reply:
In short: if it is possible to get:
https://github.com/macports/macports-ports/blob/7d3cdee471ccc3514bea2eb1686c975249578a06/_resources/port1.0/group/boost-1.0.tcl#L52
without adding boost PG to gdal+libkml, then that would likely be enough.